doc/spinel-protocol-src/spinel-tech-thread.md

```diff
@@ -287,5 +287,65 @@ CoAP requests/responses from the NCP's RLOC address. This allows the
 host driver to implement a Thread border agent.
 
 
+### PROP 5395: PROP_THREAD_DISOVERY_REQUEST_JOINER_FLAG {#prop-thread-discovery-request-joiner-flag}
 
+* Type: Read-Write
+* Packed-Encoding:: `b`
+
+This property specifies the value used in Thread MLE Discovery Request
+TLV during discovery scan operation. Default value is `false`.
+
+### PROP 5396: PROP_THREAD_DISCOVERY_SCAN_ENABLE_FILTERING {#prop-thread-discovery-scan-enable-filtering}
+
+* Type: Read-Write
+* Packed-Encoding:: `b`
+
+This property is used to enable/disable EUI64 filtering during discovery
+scan operation. Default value is `false`.
+
+### PROP 5397: PROP_THREAD_DISCOVER_SCAN_RESULT (#prop-thread-discover-scan-result)
+
+* Type: Signal
+* Packed-Encoding:: `CcdD
+
+This property is emitted asynchronously via `CMD_PROP_VALUE_IS` to
+provide a scan result during discovery scan operation.
+
+Octets: | 1  |   1  |    2    |   *n*    |    2    |   *n*
+--------|----|------|---------|----------|---------|----------
+Fields: | CH | RSSI | MAC_LEN | MAC_DATA | NET_LEN | NET_DATA
+
+Discover result have two embedded structures which contain
+information about the MAC layer and the NET layer.
+The format below is for discover result:
+
+* `C`: Channel
```

Comment:
It's the same approach that we talked about. We add a new scanning mode `SPINEL_SCAN_STATE_DISCOVER` which starts the discover scan operation. The discover operation uses/shares the parameter `SPINEL_PROP_MAC_SCAN_MASK` to decide the channel mask and also uses the newly added `PROP_THREAD_DISOVERY_REQUEST_JOINER_FLAG` and `PROP_THREAD_DISCOVERY_SCAN_ENABLE_FILTERING` as input argument to configure the discover scan operation.

Added the property `PROP_THREAD_DISCOVER_SCAN_RESULT` to emit the discovery scan result. The data items in this property are almost the same as `SPINEL_PROP_MAC_SCAN_BEACON` but with discover result we also have steering data as part of net layer properties which does not really apply in case of active scan.

The key question is if we want to reuse the `SCAN_BEACON` for both active and discover scan result or define a new one like `PROP_THREAD_DISCOVER_SCAN_RESULT`. Both ways seem ok to me. Slight upside of defining a new property is that it can help us distinguish the result and we can in future add new sub-property items to either one independently.